### PR TITLE
Fix random test failures

### LINKED ISSUE #615: Random build failures on Jenkins
See https://ci.centos.org/job/devtools-fabric8-auth/1546/consoleFull:
````
--- FAIL: TestRunInvitationREST (1.27s)
    --- FAIL: TestRunInvitationREST/TestRescindInvitation (1.01s)
    	assertions.go:247:                           
	Error Trace:	space_wrapper.go:45    			            	
			test_graph.go:74    			            	
			test_graph.go:96    			            	
			team_wrapper.go:52    			            	
			team_wrapper.go:52    			            	
			test_graph.go:74    			            	
			test_graph.go:156    			            	
			invitation_blackbox_test.go:294    		
	Error:      	Received unexpected error:    		
	            	pq: deadlock detected    		
	            	github.com/fabric8-services/fabric8-auth/authorization/role/repository.(*GormRoleMappingRepository).Create    		
	            		/tmp/go/src/github.com/fabric8-services/fabric8-auth/authorization/role/repository/role_mapping.go:111
````

and https://ci.centos.org/job/devtools-fabric8-auth/1545/consoleFull

````
--- FAIL: TestRunCollaboratorsREST (2.07s)
    --- FAIL: TestRunCollaboratorsREST/TestRemoveManyCollaboratorsWithWrongUserIDFormatReturnsBadRequest (1.01s)
    	space_testing.go:285: invalid response status code: got 500, expected 200
    	space_testing.go:292: invalid response media: got variable of type *app.JSONAPIErrors, value &{Errors:[0xc4200e9cc0]}, expected instance of app.SpaceResource
````

### PR DESCRIPTION
**Pull request by xcoulon**
refactor TestInvitationController and TestCollaboratorsController to use subtests
 with a new test graph for each test.

Fixes #615

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-616-1`

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-616-2`

**Comment by sbose78** [comment]
[test]

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-616-3`

**Comment by sbose78** [comment]
@xcoulon could you please point out exactly what is different apart from the reorganizing the tests? Is it the test graph usage?

**Comment by xcoulon** [comment]
>could you please point out exactly what is different apart from the reorganizing the tests? Is it the test graph usage?

@sbose78 yes, as the description of the PR explains:

> refactor TestInvitationController and TestCollaboratorsController to use subtests
with a new test graph for each test.

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-616-5`

File: account/repository/identity_blackbox_test.go [hunk 1]
```diff
@@ -16,317 +15,335 @@ import (
...
+			Cluster:  "https://api.starter-us-east-2a.openshift.com",
 		}
-	}
+		testIdentity := &repository.Identity{
+			Username:     "TestLoadIdentityAndUserOK" + uuid.NewV4().String(),
+			ProviderType: repository.KeycloakIDP,
+			User:         *testUser,
+		}
+		userRepository := repository.NewUserRepository(s.DB)
+		userRepository.Create(s.Ctx, testUser)
```
Comment (golangci) [inline account/repository/identity_blackbox_test.go]: Error return value of `userRepository.Create` is not checked

File: account/repository/identity_blackbox_test.go [hunk 2]
```diff
@@ -16,317 +15,335 @@ import (
...
+		s.Application.Identities().Create(s.Ctx, testIdentity)
```
Comment (golangci) [inline account/repository/identity_blackbox_test.go]: Error return value of `(github.com/fabric8-services/fabric8-auth/account/repository.IdentityRepository).Create` is not checked

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-616-6`

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-616-7`

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-616-9`

**Comment by xcoulon** [comment]
thanks, @sbose78 !

**Pull request merged**